native/src/init/lib.rs

```diff
@@ -1,11 +1,87 @@
 pub use base;
+use base::read_lines;
 pub use logging::*;
+use std::env;
+use std::path::Path;
 
 mod logging;
 
 #[cxx::bridge(namespace = "rust")]
 pub mod ffi2 {
     extern "Rust" {
         fn setup_klog();
+
+        fn rules_device_config() -> i32;
+    }
+}
+
+pub fn rules_device_config() -> i32 {
+    const UNKNOWN: i32 = 0;
+    const PERSIST: i32 = UNKNOWN + 1;
+    const METADATA: i32 = PERSIST + 1;
+    const CACHE: i32 = METADATA + 1;
+    const UNENCRYPTED: i32 = CACHE + 1;
+    const DATA: i32 = UNENCRYPTED + 1;
+    const OLD: i32 = DATA + 1;
+
+    let encrypted = env::var("ISENCRYPTED").map_or(false, |var| var == "true");
```

Comment:
This PR was brought to my attention, and I just wanted to mention that this line isn't very Rust-like at all. There are better approaches to this code, and I thought I'd leave an amendment below:

```
let encrypted = env::var("ISENCRYPTED").unwrap_or(String::from("false"));
```

Presuming you want the default value to be "false". If not, you can change `String::from("false")` to `String::from("true")`.

You can then use traditional comparison operators on the `String`, or `.eq(T)`. 

I really like Magisk, so this is my way of just helping out with Rust side of things.

Reply:
Having looked it again, I see something else. You're comparing it to a boolean. I didn't see that part clearly, it was nested. You could use `bool::from_str`, but I'm not convinced using `..map_or` is very Rusty. You're using `.map_or` to get a default value, right?

Reply:
@shymega 
1. `encrypted` is bool, but `env::var("ISENCRYPTED").unwrap_or(String::from("false"))` is `String`, and thus it's not correct.
2. if use `bool::from_str`, I have to `bool::from_str(&env::var("r").unwrap_or(String::from("false"))[..]).unwrap_or(false)`. I don't think this verbose way is better.

Reply:
@yujincheng08 Yeah, I got the wrong impression. I think your solution, now I have re-read the PR, works fine. My apologies for the misunderstanding - your way is cleaner.